src/components/AssetsTableContent.jsx

```diff
@@ -79,37 +78,27 @@ const AssetsTableContent = (props) => {
                 <CategoryContainer />
               </ModalComponent>
             </Table.HeaderCell>
-            <Table.HeaderCell>
-              Action
-            </Table.HeaderCell>
           </Table.Row>
         </Table.Header>
 
         <Table.Body>
           {
             props.activePageAssets.map(asset => (
               <TableRowComponent
+                handleViewClick={() => props.handleViewAsset(asset.serial_number)}
```

Comment:
Please don't use a curried function here, as a new function will be created each time the render function is executed. If you find yourself doing this, then it is a cue to break down the logic or create a new component (the exception is the new `render` prop). Therefore, change [TableRowComponent](https://github.com/AndelaOSP/art-dashboard/blob/develop/src/components/TableRowComponent.jsx) from functional to a class component and add the `handleViewClick` method. Pass whatever you need as prop.

Here is an [interesting read](https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36).

Reply:
Thanks!